xwiki-platform-core/xwiki-platform-panels/xwiki-platform-panels-ui/src/main/resources/Panels/MyRecentModifications.xml

```diff
@@ -173,15 +173,36 @@
     <property>
       <content>{{velocity}}
 #set ($void = $services.async.useEntity("wiki:$xcontext.database"))
-#set ($query = $services.query.xwql('where doc.author = :author order by doc.date desc'))
+#set ($query = $services.query.hql('select doc.fullName, doc.language from XWikiDocument doc where doc.author = :author order by doc.date desc'))
 #set ($recentDocs = $query.addFilter('hidden').addFilter('unique').bindValue('author', $xcontext.user).setLimit(5).execute())
+## We count the occurence of a document to specify the document language when there are duplicates
+#set ($docNamesCounter = {})
+#foreach ($doc in $recentDocs)
+  #if ($docNamesCounter.containsKey($doc[0]))
+    #set ($docNamesCounter[$doc[0]] = $docNamesCounter[$doc[0]] + 1)
+  #else
+    #set ($docNamesCounter[$doc[0]] = 1)
+  #end
+#end
 #if ($recentDocs.size() &gt; 0 || $showEmptyPanels)
   #panelheader($services.localization.render('xe.panels.modifications.my'))
-  #foreach ($docName in $recentDocs)
-    #set ($recentDoc = $xwiki.getDocument($docName).getTranslatedDocument())
-    #if ($recentDoc)
-      ## We use HTML here because we don't have a tool to escape wiki syntax in document title.
-      * {{html}}&lt;a href="$recentDoc.getURL()"&gt;$escapetool.xml($recentDoc.plainTitle)&lt;/a&gt;{{/html}}
+  #foreach ($doc in $recentDocs)
+    #if ($docNamesCounter[$doc[0]] &gt; 1)
+      #set ($recentDoc = $xwiki.getDocument($doc[0]).getTranslatedDocument($doc[1]))
+      #if ($recentDoc)
+        #set ($docLanguage = $recentDoc.language)
+        #if ("$docLanguage" == '')
+          #set ($docLanguage = $recentDoc.defaultLanguage)
+        #end
+        ## We use HTML here because we don't have a tool to escape wiki syntax in document title.
+        * {{html}}&lt;a href="$recentDoc.getURL('view', "language=${docLanguage}")"&gt;$escapetool.xml($recentDoc.plainTitle) &lt;em&gt;($docLanguage)&lt;/em&gt;&lt;/a&gt;{{/html}}
+      #end
+    #else
+      #set ($recentDoc = $xwiki.getDocument($doc[0]).getTranslatedDocument())
```

Comment:
I don't agree with this logic. Having only one entry does not mean that entry is the one corresponding to the context locale.

Reply:
I thought it might be a good idea to keep the actual behavior and only change it when we have duplicates.
I guess we can decide to always use the document with the contextual language.